### PR TITLE
Add regenerated ncgen flex/bison output files + genparser script

### PR DESCRIPTION
**Pull request by derobins**
The flex/bison-generated files are not routinely generated when the input files change. This brings the genparser script from HDF5 to make running flex/bison easier so we start doing this.

I did not change the input files to eliminate warnings. I will do that in a later PR.

### CODE REVIEW

**Comment by derobins** [comment]
I did not change any of the input files so the output changes are due to the flex/bison version changes (which itself could fix bugs) or changes to the input files that were made but never propagated since we didn't routinely regenerate the files.

File: bin/genparser [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#!/bin/sh
+#
+# Copyright by The HDF Group.
+# Copyright by the Board of Trustees of the University of Illinois.
```
Comment (lrknox) [inline bin/genparser:4]: Just checking:  Do you want to include this Copyright line?

**Review by lrknox — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

File: bin/genparser (hunk 1 above)
Comment (derobins) [reply bin/genparser:4]: I'm just copying the HDF4 lines. Do we need to apply the HDF5 rules to HDF4?

Comment (bmribler) [inline bin/genparser:4]: You meant the UofI line?

Comment (bmribler) [reply bin/genparser:4]: > Just checking: Do you want to include this Copyright line?

We need to keep the UofI copyright line if the file was created before 2006.

**Pull request merged**